src/App.ts

```diff
@@ -63,6 +63,14 @@ export interface AppOptions {
   signingSecret?: ExpressReceiverOptions['signingSecret'];
   endpoints?: ExpressReceiverOptions['endpoints'];
   processBeforeResponse?: ExpressReceiverOptions['processBeforeResponse'];
+  clientId?: ExpressReceiverOptions['clientId'];
+  clientSecret?: ExpressReceiverOptions['clientSecret'];
+  stateStore?: ExpressReceiverOptions['stateStore']; // default ClearStateStore
+  stateSecret?: ExpressReceiverOptions['stateSecret']; // required when using default stateStore
+  installationStore?: ExpressReceiverOptions['installationStore']; // default MemoryInstallationStore
+  authVersion?: ExpressReceiverOptions['authVersion']; // default 'v2'
+  scopes?: ExpressReceiverOptions['scopes'];
```

Comment:
Need to support the options for `userScopes` and `redirectUri`.

Reply:
As @aoberoi and I discussed this, we seemed to be leaning towards the bolt implementation not offering everything that the OAuth package includes. The idea being that if folks wanted to set `userScopes` they would have to use the `OAuth` package directly. Same goes for `redirect_uri` (well they could also just set it on their app config page). 

What are your thoughts on this @seratch? Would it be better to include those options in bolt instead?

Reply:
My hypothesis is that requesting user scopes is a minority use case.

But it remains to be seen whether it’s a small enough minority that it does not warrant explicit support from the `App` API. For developers who need to use user scopes, we can try recommending `app.receiver.installer.generateInstallUrl()`. It should be possible for us to add `userScopes` as an option if we receive feedback that it would be helpful. The cost of adding it now is that it might “light the path” towards a workflow we don’t think is ideal (in general, user scopes should be avoided if possible). 

I think using `redirectUrl` is similarly a minority use case. As long as your app uses just one redirect URL, you can set this in your app configuration. The option would be for developers who have distinct redirect URLs for different occurrences of the install flow. In this use case, I’d rather recommend that developers use `app.receiver.installer.generateInstallUrl()` since this will allow them to change the redirect URL (and other options) per occurrence. Similar to above, if we receive feedback that it would be helpful to add the option, we can later. There’s much less risk of lighting up a non-ideal path for this option, but also less utility of creating an explicit option. 

Lastly, I’m a little bit concerned about the amount of new options we are adding to the `App` API. The ones we’ve chosen are all useful and feel necessary. But should we group some of the less often used ones into a subproperty (like `clientOptions`), to “hide” the ones we feel developers will use less frequently? I’m imagining producing reference docs for `AppOptions` (hopefully soon) and it could look overwhelming with this many options. What do you all think?

Reply:
> subproperty

I'm very supportive of this idea. Also, if we go with this, probably there is no reason to ignore those two minor options in it.

Reply:
But it's also fine to hold off adding the two options until receiving users' requests.

Reply:
Added `installerOptions`. Holding off on adding `redirect_uri` and `userScopes` until it is requested.

Reply:
Got a request: https://github.com/slackapi/bolt-js/issues/478#issuecomment-633457952